### PR TITLE
Support new soc NXP S32Z280

### PR DESCRIPTION
**Pull request by haduongquang**
This pull request adds support for new S32Z280 soc based on the S32Z270.

### CODE REVIEW

File: soc/nxp/s32/s32ze/Kconfig.soc [hunk 1]
```diff
@@ -14,19 +14,40 @@ config SOC_S32Z270
...
 	select SOC_S32Z270
 
+config SOC_S32Z280_RTU1
+	bool
+	select SOC_S32Z280
+
 config SOC
 	default "s32z270" if SOC_S32Z270
 
+config SOC
```
Comment (nordicjm) [inline soc/nxp/s32/s32ze/Kconfig.soc:40]:
```suggestion
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/nxp/s32/s32ze/Kconfig.soc [hunk 2]
```diff
@@ -14,19 +14,40 @@ config SOC_S32Z270
...
 	bool
 
+config SOC_PART_NUMBER_P32Z280ABCMJF
+	bool
+
 config SOC_PART_NUMBER
 	default "P32Z270ADCK0MJFT" if SOC_PART_NUMBER_P32Z270ADCK0MJFT
+
+config SOC_PART_NUMBER
+	default "P32Z280ABCMJF" if SOC_PART_NUMBER_P32Z280ABCMJF
```
Comment (manuargue) [inline soc/nxp/s32/s32ze/Kconfig.soc:53]:
```suggestion
config SOC_PART_NUMBER
	default "P32Z270ADCK0MJFT" if SOC_PART_NUMBER_P32Z270ADCK0MJFT
	default "P32Z280ABCMJF" if SOC_PART_NUMBER_P32Z280ABCMJF
```

**Review by manuargue — CHANGES_REQUESTED** [review]
is there any way we can reuse dts overlays between Z280 and Z270? it's the same board, just different chip, so I'd expect pretty much everything can be reused between these two socs and the board revisions B and D

File: soc/nxp/s32/s32ze/Kconfig.soc (hunk 2 above)
Comment (haduongquang) [reply soc/nxp/s32/s32ze/Kconfig.soc]: I updated

File: soc/nxp/s32/s32ze/Kconfig.soc (hunk 1 above)
Comment (haduongquang) [reply soc/nxp/s32/s32ze/Kconfig.soc]: I updated

File: boards/nxp/s32z2xxdc2/s32z2xxdc2_s32z280_rtu0_defconfig [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_XIP=n
+CONFIG_ISR_STACK_SIZE=512
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=8000000
+CONFIG_ARM_MPU=y
+CONFIG_PINCTRL=y
```
Comment (gmarull) [inline boards/nxp/s32z2xxdc2/s32z2xxdc2_s32z280_rtu0_defconfig:9]: please, do not enable pinctrl at board level, let drivers who need it select the option.

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/nxp/s32z2xxdc2/s32z2xxdc2_s32z280_rtu0_defconfig (hunk 3 above)
Comment (haduongquang) [reply boards/nxp/s32z2xxdc2/s32z2xxdc2_s32z280_rtu0_defconfig:9]: Hello, i can remove this at here. But looking around in-tree boards, various boards have this Kconfig option enabled. So I'm wondering if we have this guideline documented somewhere for people who introduce new boards to be aware?

**Comment by haduongquang** [comment]
Hi all, I will temporarily close this PR, and re-open it when necessary.

**Pull request closed without merging**